src/multicellular_stage/systems/IntercellularMatrixSystem.cs

```diff
@@ -0,0 +1,150 @@
+        instance.AddChild(connection);
+
+        // Get target's relative position (taking rotation into account) using inverse transform multiplication
+        var inverseColonyTransform = entity.Get<WorldPosition>().ToTransform().Inverse();
+        var targetRelativePos = inverseColonyTransform * parentEntity.Get<WorldPosition>().Position;
+
+        Vector3 pointA, pointB;
+        (pointA, pointB) = FindGoodConnectionPoints(ourMembrane.MembraneData,
+            targetMembrane.MembraneData, targetRelativePos);
+
+        var relativePosition = pointB - pointA;
+        float relativePosLength = relativePosition.Length();
+
+        var angle = relativePosition.AngleTo(Vector3.Forward);
+        if (relativePosition.X > 0.0f)
+            angle *= -1.0f;
+
+        connection.Scale = new Vector3(5.0f, 1.0f, relativePosLength + 3.0f);
+        connection.RotateY(angle);
+        connection.Position += (pointA + pointB) * 0.5f;
+
+        intercellularMatrix.GeneratedConnection = connection;
+
+        ApplyConnectionMaterialParameters(entity, ref intercellularMatrix);
+    }
+
+    private static (Vector3 PointA, Vector3 PointB) FindGoodConnectionPoints(MembranePointData membraneA,
+        MembranePointData membraneB, Vector3 membraneBOffset)
+    {
+        Vector2 closestVertexB = FindClosestVertex(Vector2.Zero, membraneB.Vertices2D,
+            new Vector2(membraneBOffset.X, membraneBOffset.Z));
+
+        Vector2 closestVertexA = FindClosestVertex(closestVertexB, membraneA.Vertices2D, Vector2.Zero);
+
+        return (new Vector3(closestVertexA.X, 0.0f, closestVertexA.Y),
+            new Vector3(closestVertexB.X, 0.0f, closestVertexB.Y));
+    }
+
+    private static Vector2 FindClosestVertex(Vector2 target, Vector2[] vertices, Vector2 offset)
+    {
+        Vector2 closestVertex = Vector2.Zero;
+        float minDistance = float.MaxValue;
+        foreach (var vertex in vertices)
+        {
+            var relativeVertex = vertex + offset;
+
+            float squareDistance = relativeVertex.DistanceSquaredTo(target);
+
+            if (squareDistance < minDistance)
+            {
+                closestVertex = relativeVertex;
+                minDistance = squareDistance;
+            }
+        }
+
+        return closestVertex;
+    }
+
+    private static void RemoveConnection(ref IntercellularMatrix intercellularMatrix)
+    {
+        intercellularMatrix.GeneratedConnection?.QueueFree();
+        intercellularMatrix.GeneratedConnection = null;
+    }
+
+    private static void ApplyConnectionMaterialParameters(in Entity entity,
+        ref IntercellularMatrix intercellularMatrix)
+    {
+        if (intercellularMatrix.GeneratedConnection == null)
+        {
+            GD.PrintErr("Intercellular connection is null, can't apply material parameters");
+            return;
+        }
+
+        TempMaterialList.Clear();
+        intercellularMatrix.GeneratedConnection.GetMaterial(TempMaterialList);
```

Comment:
GetMaterial is our helper method, but in this situation where the exact visual instance that is created, I think it would be better to directly read the data from Godot. 

Also with the current `TempMaterialList` there is a thread race condition problem here.